sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryPolicy.java

```diff
@@ -36,4 +38,8 @@ public DocumentClientRetryPolicy getRequestPolicy() {
     public RetryContext getRetryContext() {
         return null;
     }
+
+    public void setRxCollectionCache(RxCollectionCache rxCollectionCache) {
```

Comment:
why can't we pass this down from the constructor? 

that would prevent forgetting to invoke the setter.

Reply:
Please see this comment https://github.com/Azure/azure-sdk-for-java/pull/25811#discussion_r764327955

Reply:
Discussed offline, there will be no race condition , because this is getting set only one time inside client initialization method